js/data/bucket/symbol_bucket.js

```diff
@@ -218,16 +218,28 @@ SymbolBucket.prototype.populateArrays = function(collisionTile, stacks, icons) {
         textFeatures = merged.textFeatures;
     }
 
-    var shapedText, shapedIcon;
+    var shapedText, shapedTextVertical, shapedIcon;
 
     for (var k = 0; k < features.length; k++) {
         if (!geometries[k]) continue;
 
+        var verticalOrientation = false;
+
         if (textFeatures[k]) {
+
             shapedText = shapeText(textFeatures[k], stacks[fontstack], maxWidth,
-                    lineHeight, horizontalAlign, verticalAlign, justify, spacing, textOffset);
+                    lineHeight, horizontalAlign, verticalAlign, justify, spacing, textOffset, oneEm, verticalOrientation);
+
+            if (layout['text-rotation-alignment'] === 'map' && layout['symbol-placement'] === 'line') {
```

Comment:
To avoid verticalizing non-CJK* text, match `textFeatures[k]` against this regular expression: `/[ᄀ-ᇿ가-힣ㄱ-ㆎ一-鿌㐀-䶵　-〿𠀀-𬺯！-￮ぁ-ゟꀀ-꓆᠀-ᢪ]/`.

<sub>* For the purposes of this PR, “CJK” is Hangul, Hanzi, Hiragana, Mongolian, and Yi scripts. However, note that Hangul and Mongolian are separated by spaces and thus should retain the Latin-style line breaking algorithm.</sub>